engine/src/lib.rs

```diff
@@ -105,6 +105,7 @@ mod contract {
     const CODE_KEY: &[u8; 4] = b"CODE";
     const CODE_STAGE_KEY: &[u8; 10] = b"CODE_STAGE";
 
+    #[allow(clippy::empty_line_after_doc_comments)]
```

Comment:
Would it be better to remove this empty line?

Reply:
Actually, it's not `doc comments` at all. It's comments for codec section.